### PR TITLE
Add other category to dev portfolio form

### PR DESCRIPTION
**Pull request by patriciaahuang**
### Summary

The dev portfolio submissions form now has a third category besides “Opened PRs” and “Reviewed PRs” to handle exceptions that don’t fall into either category, but should be counted as valid. This category is automatically hidden unless the user explicitly clicks the arrow to open this section. 

### Notion Link

https://www.notion.so/DP-Add-Other-Category-to-Allow-Exceptions-7ed09a2923ea4a51975ed773de0bcfd8?pvs=4

### Test Plan

I checked that opening the third category will set the previous two categories to optional.

Screen Shot 2023-11-25 at 3 40 20 AM

I created a test dev portfolio that used the other category, and checked the submissions dashboard to see that the submission is automatically set to pending.
Screen Shot 2023-11-25 at 3 39 54 AM

The PRs and explanation are shown when the admin opens the "Show Text" button.
Screen Shot 2023-11-25 at 3 40 01 AM

### CODE REVIEW

File: frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx
Comment (andrew032011) [inline frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx:82]: Rewrite the condition as `!isTpm && otherEmpty && (openedEmpty || reviewedEmpty)` and the delete the inner if-statement. If you try it out, this logic doesn't allow you to submit regular Dev Portfolios without otherPRs.

File: frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx
Comment (andrew032011) [inline frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx:375]: I'd argue that `OtherPRInputs` doesn't apply to TPMs and shouldn't be shown at all to TPMs. Since TPMs don't have an opened/reviewed PRs requirement anyway (i.e. they can submit every dev portfolio without any reviewed/opened PRs), the TPM can just put extra PRs that don't apply under the "what did I work on" section. 

In fact, we're using the same `text` field on the DB level as the TPM one, and I see that they are currently managed as the same state on the frontend (`text, setText`), so there's a conflict there we'd have to address.

Hiding this for TPMs will save you that hassle.

File: frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx
Comment (andrew032011) [inline frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx:314]: I really like that you're thinking about the UX here and hiding the asterisk when the user opens up the other PR's field 👍.

**Review by andrew032011 — CHANGES_REQUESTED** [review]
Great work. Some considerations on edge-case handling.

Let me know also what you think about making the `openedPRs`, `reviewedPRs`, `otherPRs` fields non-optional. I think it'd save a lot of `undefined`-checks throughout the code.

Also, because we're changing the DB schema, even though we're just adding a field, let's test the schema change on Dev DB, and once you're sure that your script doesn't break anything else, we can run it once on the prod DB right before merging it in.

File: frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx
Comment (patriciaahuang) [reply frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx:82]: Thanks for pointing this out! I had a misunderstanding that if otherPRs is empty, then both an opened PR and a reviewed PR had to be submitted.

File: frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx
Comment (andrew032011) [reply frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx:375]: Yeah, you can remove those! Hm, I still see "Other PRs" on TPM view.
Screenshot 2023-11-27 at 11 23 14 AM

File: backend/src/utils/githubUtil.ts
Comment (andrew032011) [inline backend/src/utils/githubUtil.ts:297]: We can remove the falsy-check here, since we know the list will be defined.

File: frontend/src/components/Admin/DevPortfolio/DevPortfolioDetails.tsx
Comment (andrew032011) [inline frontend/src/components/Admin/DevPortfolio/DevPortfolioDetails.tsx:224]: Remove falsy check here.

File: frontend/src/components/Modals/DevPortfolioTextModal.tsx
Comment (andrew032011) [inline frontend/src/components/Modals/DevPortfolioTextModal.tsx:56]: Let's resolve this warning:

Replace

```
<a>
```

with

```
<a href={prSubmission.url} target="_blank" rel="noreferrer noopener">
```

**Review by andrew032011 — APPROVED** [review]
Minor nits. Feel free to merge this in after resolving.

**Review by andrew032011 — APPROVED** [review]

**Comment by andrew032011** [comment]
CLA check was hanging for some reason. I ran 

```
https://cla-assistant.io/check/cornell-dti/idol?pullRequest=553
```

to rerun it. From looking around some other PR's were affected too. Will keep an eye to see if it keeps happening.

**Pull request merged**
